dashboard/test/ui/features/step_definitions/steps.rb

```diff
@@ -1032,6 +1048,34 @@ def set_cookie(key, value)
   @browser.manage.add_cookie params
 end
 
+def set_cookie_dcdo(key, value)
+  cookie_dcdo =
+    begin
+      JSON.parse(@browser.manage.cookie_named('DCDO').try(:[], :value).presence || '{}')
+    rescue Selenium::WebDriver::Error::NoSuchCookieError
+      {}
+    end
+
+  cookie_dcdo[key] = value
+
+  begin
+    set_cookie('DCDO', cookie_dcdo.to_json)
+  rescue Selenium::WebDriver::Error::InvalidCookieDomainError
+    navigate_to replace_hostname('http://studio.code.org')
+    retry
+  end
+end
+
+Given(/^I set the DCDO flag "([^"]*)" to "(.*)"$/) do |key, json|
```

Comment:
nit-pick to match language in `dynamic_config_base.rb`
```suggestion
Given(/^I set the DCDO key "([^"]*)" to "(.*)"$/) do |key, json|
```